### PR TITLE
Same scope as mx.fast.rms_norm this adds mx.fast.layer_norm

### PR DESCRIPTION
**Pull request by angeloskath**
Starting the PR for review and will be adding some benchmarks as I am running them.

### CODE REVIEW

File: mlx/fast.cpp [hunk 1]
```diff
@@ -87,7 +87,7 @@ array rms_norm(
   if (s.device == Device::gpu) {
     return array(
         x.shape(),
-        x.dtype(),
+        out_type,
```
Comment (awni) [inline mlx/fast.cpp:90]: that looks like a bug.. 😬

Comment (angeloskath) [reply mlx/fast.cpp:90]: Looks like a bug now or before? Cause I thought I was fixing a bug :-)

Comment (awni) [reply mlx/fast.cpp:90]: Yea I meant before. Good catch!

**Comment by awni** [comment]
`python -m mlx_lm.generate --model  mlx-community/starcoder2-7b-4bit --prompt "Write a quick sort in C++" --max-tokens 256 --temp 0.0`

Pre
```
Generation: 67.713 tokens-per-sec
```

Post
```
Generation: 75.267 tokens-per-sec
```

**Review by awni — APPROVED** [review]
🚀

**Comment by angeloskath** [comment]
Waiting on the tests and merging it.

**Pull request merged**